src/sage/rings/polynomial/integer_valued_polynomials.py

```diff
@@ -0,0 +1,1125 @@
+                    sage: S[5].delta()
+                    S[0] + S[1] + S[2] + S[3] + S[4]
+                """
+                return self.variable_shift() - self
+
+            def variable_shift(self, k=1):
+                r"""
+                Return the image by the shift of variables.
+
+                On polynomials, the action is the shift
+                on variables `x \mapsto x + 1`.
+
+                INPUT:
+
+                - `k` -- integer (default: 1)
+
+                EXAMPLES::
+
+                    sage: A = IntegerValuedPolynomialRing(ZZ).S()
+                    sage: S = A.basis()
+                    sage: S[5].variable_shift()
+                    S[0] + S[1] + S[2] + S[3] + S[4] + S[5]
+                """
+                if k == 0:
+                    return self
+
+                A = self.parent()
+                resu = A.sum(c * A.monomial(j) for i, c in self
+                             for j in range(i + 1))
+                if k == 1:
+                    return resu
+                return resu.variable_shift(k - 1)
+
+            def variable_unshift(self, k=1):
+                r"""
+                Return the image by the unshift of variables.
+
+                On polynomials, the action is the shift
+                on variables `x \mapsto x - k`.
+
+                INPUT:
+
+                - `k` -- integer (default: 1)
+
+                EXAMPLES::
+
+                    sage: A = IntegerValuedPolynomialRing(ZZ).S()
+                    sage: S = A.basis()
+                    sage: S[5].variable_unshift()
+                    -S[4] + S[5]
+                """
+                if k == 0:
+                    return self
+
+                A = self.parent()
+                resu = (A.sum(c * A.monomial(i) for i, c in self) -
+                        A.sum(c * A.monomial(i - 1) for i, c in self if i))
+                if k == 1:
+                    return resu
+                return resu.variable_unshift(k - 1)
+
+            def derivative_at_minus_one(self):
+                """
+                Return the derivative at `-1`.
+
+                This is sometimes useful when `-1` is a root.
+
+                EXAMPLES::
+
+                    sage: F = IntegerValuedPolynomialRing(ZZ).S()
+                    sage: B = F.gen()
+                    sage: (B+1).derivative_at_minus_one()
+                    1
+                """
+                return QQ.sum(c / QQ(i) for i, c in self if i)
+
+            special_value = derivative_at_minus_one
+
+            def polynomial(self):
+                """
+                Convert to a standard polynomial in `x`.
+
+                EXAMPLES::
+
+                    sage: F = IntegerValuedPolynomialRing(ZZ).S()
+                    sage: B = F.gen()
+                    sage: (B+1).polynomial()
+                    x + 2
+                """
+                x = polygen(QQ, 'x')
+                return sum(c * binomial(x + i, i) for i, c in self)
```

Comment:
This will have a different result for `0`. Also add a test for this corner case checking for a consistent parent.